lib/test/openapi_assert_helper.ex

```diff
@@ -0,0 +1,695 @@
+      end
+    end
+
+    defunp pick_request_body(asserter :: SchemaAsserter.t()) :: v[map | nil] do
+      case asserter.api["requestBody"] do
+        %{"content" => %{"application/json" => %{"schema" => schema}}} ->
+          schema
+
+        _ ->
+          nil
+      end
+    end
+
+    defunp success_response_status(asserter :: SchemaAsserter.t()) :: v[non_neg_integer] do
+      {status, _} = pick_success_response(asserter)
+      status
+    end
+
+    defunp pick_success_response(asserter :: SchemaAsserter.t()) ::
+             v[{non_neg_integer, map | nil}] do
+      responses = asserter.api["responses"]
+
+      success_key =
+        responses
+        |> Map.keys()
+        |> Enum.find(fn key ->
+          # credo:disable-for-next-line Credo.Check.Refactor.AppendSingleItem
+          key in (200..207 |> Enum.map(&to_string/1))
+        end)
+
+      {String.to_integer(success_key),
+       responses[success_key]["content"]["application/json"]["schema"]}
+    end
+
+    defunp validate_response_body(
+             asserter :: SchemaAsserter.t(),
+             body :: v[map | list | String.t()]
+           ) ::
+             v[:ok | {:error, map | String.t()}] do
+      case pick_success_response(asserter) do
+        {_, nil} ->
+          if body == "" do
+            :ok
+          else
+            {:error, "schema is nothing, but response body is not empty"}
+          end
+
+        {_, schema} ->
+          with_ref_target =
+            Map.merge(schema, %{
+              "components" => asserter.components
+            })
+
+          resolved_schema = ExJsonSchema.Schema.resolve(with_ref_target)
+          schema_without_additional_properties = resolve_reference_if_all_of(resolved_schema)
+
+          ExJsonSchema.Validator.validate(schema_without_additional_properties, body,
+            error_formatter: false
+          )
+      end
+    end
+
+    # $allOf with item having additionalProperties:false makes validation error.
+    # ref. https://github.com/getkin/kin-openapi/issues/101
+    # So, this resolves $ref and adds additionalProperties:false to allOf itself.
+    #
+    # This function is intentionally `defp` because the caller uses `ExJsonSchema.Schema.resolve/1`,
+    # `resolve/1` has `no_return` and it breaks `resolved_schema :: ExJsonSchema.Schema.Root`
+    # credo:disable-for-next-line Credo.Check.Refactor.ABCSize
+    defp resolve_reference_if_all_of(resolved_schema) do
+      schema = resolved_schema.schema
+
+      if Map.has_key?(schema, "allOf") do
+        ref_targets =
+          schema["allOf"]
+          # e.x. item %{"$ref" => [:root, "components", "schemas", "AdminUser"]},
+          |> Enum.filter(fn x -> Map.has_key?(x, "$ref") end)
+          |> Enum.map(fn x ->
+            [:root, "components" | rest_path] = x["$ref"]
+            rest_path
+          end)
+
+          # e.x. item ["components", "schemas", "AdminUser"]
```

Comment:
`"components"` is already dropped at above.

```suggestion
          # e.g. item ["schemas", "AdminUser"]
```

Reply:
Good catch!
Fixed at https://github.com/access-company/antikythera/pull/273/commits/aa0399d8e008cee7e8a405489b358e605e9d381a